### PR TITLE
Tidy up genetic map IDs and docs.

### PR DESCRIPTION
**Pull request by jeromekelleher**
This should finalise the changes needed in the genetic maps before release. Main changes:

- Tried to add some descriptive documentation to the maps. Users who don't know anything about the maps shouldn't have to read the paper. The long_descriptions attempt to give a summary of how the maps were made. The ``description`` is a shorter one-line description of what the map is.

- Made an attempt to make the IDs more descriptive, plus trying to avoid future name clashes (we could have the male and female maps for deCode later). I'm not sure there's much point in adding a year to the IDs since the combination of descriptive name and assembly is likely to be unique.

### CODE REVIEW

**Comment by andrewkern** [comment]
this LGTM @jeromekelleher.

**Comment by jeromekelleher** [comment]
Great, thanks @andrewkern. 

@reedacartwright, @dschride, quick opinion here: do genetic map IDs like ``HapMapII_GRCh37``,
``DeCodeSexAveraged_GRCh36``, ``ComeronCrossover_dm6`` and  ``SalomeAveraged_TAIR7`` look OK to you?

**Comment by reedacartwright** [comment]
On first pass they look good. However, the style is inconsistent with the model id scheme.  An alternative scheme is `TAIR7_Averaged_S12`, `dm6_Crossover_C12`, `GRCh37_HapMapII_H07`, and `GRCh36_DecodeSexAveraged_K10`.  I like the refernce name first because if we did a liftover to a new reference, we would be changing the first element.  Of course I could see reasons for putting it last as well.

**Comment by jeromekelleher** [comment]
> On first pass they look good. However, the style is inconsistent with the model id scheme. An alternative scheme is TAIR7_Averaged_S12, dm6_Crossover_C12, GRCh37_HapMapII_H07, and GRCh36_DecodeSexAveraged_K10. I like the refernce name first because if we did a liftover to a new reference, we would be changing the first element. Of course I could see reasons for putting it last as well.

Thanks @reedacartwright, great to get your input. Thoughts anyone else? I see the arguments here, but the current scheme feels more memorable and we're not expecting to have as many genetic maps as demographic models so we don't need to worry about collisions so much. 

I'm happy to change if there's consensus, but I'd rather not otherwise I think. Opinions either way appreciated.

**Comment by andrewkern** [comment]
I'd rather not change it at this point as we sought consensus on this issue over the past weeks and arrived at the current strategy.

**Comment by reedacartwright** [comment]
That makes sense.

File: stdpopsim/catalog/homo_sapiens.py [hunk 1]
```diff
@@ -84,38 +84,49 @@
...
-    description="Hapmap Phase II",
-    long_description=(
-        "The Phase II HapMap Genetic map (lifted over to GRCh37) used in "
-        "1000 Genomes. Please see the README for more details."),
+    id="HapMapII_GRCh37",
+    description="HapMap Phase II lifted over to GRCh37",
+    long_description="""
+        This genetic map is from the Phase II Hapmap project
+        and based on 3.1 million genotyped SNPs
+        from 270 individuals across four populations (YRI, CEY, CHB and JPT).
```
Comment (gtsambos) [inline stdpopsim/catalog/homo_sapiens.py]: Tiny typo -- this should be CEU

**Pull request merged**